homework2/task4.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define uint unsigned int
+
+#define arrElementType short
+
+#define maxNum 1000
+
+void swap(arrElementType *a, arrElementType *b)
+{
+    arrElementType tmp = *a;
+    *a = *b;
+    *b = tmp;
+}
+
+void generate(size_t len, arrElementType *arr)
+{
+    for (size_t i = 0; i < len; ++i)
+    {
+        arr[i] = rand() % maxNum;
+    }
+}
+
+void transform(size_t len, arrElementType *arr, arrElementType element)
```

Comment:
```suggestion
void partition(size_t len, arrElementType *arr, arrElementType element)
```
Более говорящее название нужно